src/lib/verifier/common.ml

```diff
@@ -1,15 +1,35 @@
 open Core_kernel
 open Mina_base
 
+type invalid =
+  [ `Invalid_keys of Signature_lib.Public_key.Compressed.Stable.Latest.t list
+  | `Invalid_signature
```

Comment:
can we add public keys for the rest of failure cases as well?

Reply:
yes, except I don't see how to add it for `Invalid_proof`, which is generated only in the batcher for the special case for a list of length one.